src/services/structures/controllers/updateDossierReconventionnement.ts

```diff
@@ -0,0 +1,113 @@
+  (app: Application) => async (req: IRequest, res: Response) => {
+    const {
+      query: { action, structureId, nombreDePostes, motif },
+      body: { conseillers },
+    } = req;
+    let statut: string;
+    let misesEnRelationObjectIds: [string];
+
+    switch (action.trim()) {
+      case 'enregistrer':
+        statut = 'ENREGISTRÉ';
+        break;
+      case 'envoyer':
+        statut = 'RECONVENTIONNEMENT_EN_COURS';
+        break;
+      case 'annuler':
+        statut = 'NON_INTERESSÉ';
+        break;
+      default:
+        res.status(400).json({ message: 'Action non valide' });
+        return;
+    }
+
+    try {
+      if (!ObjectId.isValid(structureId)) {
+        res.status(400).json({ message: 'Id incorrect' });
+        return;
+      }
+      // On modifie le statut de la structure en fonction de l'action demandée par l'utilisateur (enregistrer ou envoyer)
+      if (statut === 'ENREGISTRÉ' || statut === 'RECONVENTIONNEMENT_EN_COURS') {
+        await app
+          .service(service.structures)
+          .Model.accessibleBy(req.ability, action.update)
+          .findOneAndUpdate(
+            { _id: new ObjectId(structureId) },
+            {
+              $set: {
+                'conventionnement.statut': statut,
+                'conventionnement.derniereModification': new Date(),
+                'conventionnement.dossierReconventionnement.nbPostesAttribues':
+                  nombreDePostes,
+              },
+            },
+          );
+      } else if (statut === 'NON_INTERESSE') {
+        await app
+          .service(service.structures)
+          .Model.accessibleBy(req.ability, action.update)
+          .findOneAndUpdate(
+            { _id: new ObjectId(structureId) },
+            {
+              $set: {
+                'conventionnement.statut': statut,
+                'conventionnement.motif': motif,
+                'conventionnement.derniereModification': new Date(),
+              },
+            },
+          );
+      }
+
+      misesEnRelationObjectIds = conseillers?.map(
+        (conseiller) => new ObjectId(conseiller.miseEnRelationId),
+      );
+
+      // On modifie le statut de la mise en relation en fonction de l'action demandée par l'utilisateur (enregistrer ou envoyer)
+      await app
+        .service(service.misesEnRelation)
+        .Model.accessibleBy(req.ability, action.update)
+        .updateMany(
+          {
+            $and: [{ _id: { $in: misesEnRelationObjectIds } }],
+          },
+          { $set: { reconventionnement: true } },
+          { multi: true },
+        );
+
+      // On modifie le statut de la mise en relation en fonction de l'action demandée par l'utilisateur (enregistrer ou envoyer)
+      await app
+        .service(service.misesEnRelation)
+        .Model.accessibleBy(req.ability, action.update)
+        .updateMany(
+          {
+            $and: [
+              { _id: { $nin: misesEnRelationObjectIds } },
```

Comment:
idem ici ajouter pour sécurité un statut $in autorisées pour une telle modif

Reply:
fait